eodag/plugins/search/qssearch.py

```diff
@@ -1997,3 +1997,87 @@ def build_query_string(
     ) -> tuple[dict[str, Any], str]:
         """Build The query string using the search parameters"""
         return PostJsonSearch.build_query_string(self, product_type, query_dict)
+
+
+class GeodesSearch(StacSearch, PostJsonSearch):
+    """A specialisation of a :class:`~eodag.plugins.search.qssearch.PostJsonSearch` that uses
+    generic STAC configuration for queryables (inherited from :class:`~eodag.plugins.search.qssearch.StacSearch`).
+    Perform a second query to retrieve status of the product.
+    """
+
+    def __init__(self, provider: str, config: PluginConfig) -> None:
+        PostJsonSearch.__init__(self, provider, config)
+
+    def do_search(
+        self, prep: PreparedSearch = PreparedSearch(items_per_page=None), **kwargs: Any
+    ) -> list[Any]:
+        """
+        Perform a search on an OpenSearch-like interface and then a second query to retrieve status of the product
+        """
+        results = super().do_search(prep, **kwargs)
+
+        # Perform a second query to retrieve status of the product
+        availability_endpoint = "https://geodes-portal.cnes.fr/availability/{id}"
+        if availability_endpoint:
+            ssl_verify = getattr(self.config, "ssl_verify", True)
+            for result in results:
```

Comment:
please call availability endpoint only once with all ids, this will be way faster